src/coreclr/tools/Common/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -290,6 +290,12 @@ private void EmitPInvokeCall(PInvokeILCodeStreams ilCodeStreams)
                 fnptrLoadStream.EmitStLoc(vNativeFunctionPointer);
                 callsiteSetupCodeStream.EmitLdLoc(vNativeFunctionPointer);
                 callsiteSetupCodeStream.Emit(ILOpcode.calli, emitter.NewToken(nativeSig));
+                if (!_flags.PreserveSig)
+                {
+                    callsiteSetupCodeStream.Emit(ILOpcode.call, emitter.NewToken(
```

Comment:
Does this also need code somewhere that changes the return type of the nativeSig to be `int`?

It would be useful to add a case for this to `src\tests\nativeaot\SmokeTests\PInvoke` to validate that it works for some simple case.

Reply:
Will try to add that tomorrow.

Reply:
Could you please revert this change or address the feedback?